### PR TITLE
Adds console screens to the protolathe.

### LINKED ISSUE #16509: Add Console Screens to protolathes
#### Description of issue
Disclaimer: Paying to add code in no way guarantees a merge request will go through. At any rate, its only a dollar, so pull that stick out your butts.

I would like to request a feature, that protolathes be able to make console screens, as not being able to do so is annoying and a pain in the ass.  I have posted a bounty of one dollar, https://bountify.co/vg-station-add-feature-to-allow-protolathes-to-produce-console-screens

### PR DESCRIPTION
**Pull request by DrSnips**
Does what it says on the tin. Adds the console screens to the list of basic stock parts the protolathe can make.
Resolves #16509

### CODE REVIEW

**Comment by clusterfack** [comment]
I don't really think this autolathe specific thing makes much sense to have the protolathe produce

**Comment by DrSnips** [comment]
The proto is more or less just an advanced autolathe and it has all other stock parts to start as well.

File: code/modules/research/designs/stock_parts.dm [hunk 1]
```diff
@@ -49,6 +49,16 @@
 	category = "Stock Parts"
 	build_path = /obj/item/weapon/stock_parts/matter_bin
 
+/datum/design/console_screen
+	name = "Console Screen"
+	desc = "Used in the construction of computers and other devices with a interactive console."
```
Comment (Shadowmech88) [inline code/modules/research/designs/stock_parts.dm]: "*an* interactive console."

**Review by Shadowmech88 — CHANGES_REQUESTED** [review]

**Comment by clusterfack** [comment]
I mean not really, those were added from the protolathe to the autolathe to give it some more purpose. Which made sense. 

This however is going the other way, making the protolathe which already produces 95% of the produceable items in the game able to produce more.

**Comment by Kurfursten** [comment]
I'm with Cluster on this one, the autolathe already took a blow to relevance when console screens were added to tool storage and tech storage with the stock parts in vending machine update.

I imagine if the console screen is put into the protolathe, the compressed matter cartridge is next.

**Comment by DrSnips** [comment]
Funny you should say that about matter carts since I've been wanting to add multiple material types of matter carts to the protolathe.

**Review by MadmanMartian — COMMENTED** [review]
should make it require illegal research

**Comment by Shadowmech88** [comment]
Travis seems to have had a hiccup, I'm not sure how to restart the build.

**Review by Shadowmech88 — APPROVED** [review]

**Comment by DrSnips** [comment]
The autolathe still has it's advantages like having all designs to start as well as them being generally cheaper. To make even a beaker in the proto you need bluespace and material research done a bit as well as three times the materials. Is the beaker bigger? Yes. But two large glass beakers from the auto is still cheaper than the bluespace one from the proto.
Time wise the auto is stronger.

**Comment by DrSnips** [comment]
Let me commit something and see if that restarts it.

**Comment by clusterfack** [comment]
>having all designs to start

Which wont really matter if the protolathe starts eating all of its designs

**Comment by Kurfursten** [comment]
But you made the console screen require materials = 1 and cost trivial amounts of material and time so literally none of that applies. 🤔

**Comment by ringringllingling** [comment]
well, okay dokey, i guess you did the work, whether it went through or not, so here's your free coke.

**Comment by DrSnips** [comment]
All the other basic stock parts require the same. It's not like it should get special treatment.

**Comment by ringringllingling** [comment]
wait, didn't the console screen cost the same to make in the protolathe as the autolathe?  you should shore that up.

**Comment by ghost** [comment]
R&D is beyond unsalvageable at this point, trying to draw lines in the sand is pointless.

**Comment by DrSnips** [comment]
The cost matches the current autolathe cost. Is there something I don't see?

**Comment by clusterfack** [comment]
>R&D is beyond unsalvageable at this point, trying to draw lines in the sand is pointless.

This is the kind of ideology that corrodes the codebase as a whole

**Review by PJB3005 — APPROVED** [review]

**Comment by Exxion** [comment]
>It's already bad, so trying to keep it from getting even worse is pointless

**Comment by Aurx** [comment]
Honestly, the autolathe needs to die.  Or the protolathe, either one.  
There's simply no reason for both to exist, and trying to """balance""" them is silly when they're trying to share a very specific niche that doesn't have enough room.

**Comment by DrSnips** [comment]
Auto serves the need of immediate and cheap designs with no fuss.
Proto serves the need of complex designs that are higher in cost but also requires a lot more setup. Requiring a console, proto, and if research isnt done you need a DA and have science access to do research.
Like I said before with the beaker example.

**Comment by gbasood** [comment]
To be honest I don't see the point of keeping both around at all.
The most impact the difference has had on me is playing MoMMI on derelict, and that's about it.

**Comment by Kurfursten** [comment]
@gbasood @Aurx I don't think taking away Cargo's autolathe or giving Cargo a protolathe to replace said autolathe are good solutions. Cargo having access to an autolathe is so old and ubiquitous that every codebase has it in common.

**Comment by ringringllingling** [comment]
I don't think there is anything wrong with the autolathe or the protolathe.  They both work fine.  Its just a pain having to use the autolathe to make console screens for the mechanic or if r&d wants to build something.  Maybe remove the autolathe from R&D?

As long as the cost for the console screen in the protolathe is the same as the autolathe its all good.

**Comment by DrSnips** [comment]
Careful there. Removing the autolathe from research will likely encourage moving all autolathe designs to the proto to make up for it. I don't think that should ever be the case. Each plays a role and I've already talked about that.

**Comment by Aurx** [comment]
Eh.  It'd more just encourage R&D to build one when they wanted autolathe things.

**Comment by Aurx** [comment]
Eh.  It'd more just encourage R&D to build one when they wanted autolathe things.

**Comment by SonixApache** [comment]
console screen in auto- and protolathe is fine because they're used by different groups
R&D will likely queue up a few screens in the protolathe along with the other five billion stock parts they need
cargo / assistants / mechanics / whoever will likely get them from the autolathe since those are more easily accessible

i don't see the issue

**Comment by icantthinkofanameritenow** [comment]
In most stations, RnD already has an autolathe. It's pointless.

**Comment by FudgePucker** [comment]
Mechanics need to make an autolathe. If it wasn't for the screen in tech storage they would have to go to cargo or science to get one.

**Comment by PJB3005** [comment]
So, just make them rely on tech storage then I'd say.

**Comment by Pathid** [comment]
There is a part vendor for them in primary tool storage too.

**Pull request closed without merging**